src/careamics/lvae_training/dataset/vae_data_config.py

```diff
@@ -0,0 +1,179 @@
+from typing import Any, Optional
+from enum import Enum
+
+from pydantic import BaseModel, ConfigDict, computed_field
+
+
+# TODO: get rid of unnecessary enums
+class DataType(Enum):
+    MNIST = 0
+    Places365 = 1
+    NotMNIST = 2
+    OptiMEM100_014 = 3
+    CustomSinosoid = 4
+    Prevedel_EMBL = 5
+    AllenCellMito = 6
+    SeparateTiffData = 7
+    CustomSinosoidThreeCurve = 8
+    SemiSupBloodVesselsEMBL = 9
+    Pavia2 = 10
+    Pavia2VanillaSplitting = 11
+    ExpansionMicroscopyMitoTub = 12
+    ShroffMitoEr = 13
+    HTIba1Ki67 = 14
+    BSD68 = 15
+    BioSR_MRC = 16
+    TavernaSox2Golgi = 17
+    Dao3Channel = 18
+    ExpMicroscopyV2 = 19
+    Dao3ChannelWithInput = 20
+    TavernaSox2GolgiV2 = 21
+    TwoDset = 22
+    PredictedTiffData = 23
+    Pavia3SeqData = 24
+    # Here, we have 16 splitting tasks.
+    NicolaData = 25
+
+
+class DataSplitType(Enum):
+    All = 0
+    Train = 1
+    Val = 2
+    Test = 3
+
+
+class GridAlignement(Enum):
+    """
+    A patch is formed by padding the grid with content. If the grids are 'Center' aligned, then padding is to done equally on all 4 sides.
+    On the other hand, if grids are 'LeftTop' aligned, padding is to be done on the right and bottom end of the grid.
+    In the former case, one needs (patch_size - grid_size)//2 amount of content on the right end of the frame.
+    In the latter case, one needs patch_size - grid_size amount of content on the right end of the frame.
+    """
+
+    LeftTop = 0
+    Center = 1
+
+
+# TODO: for all bool params check if they are taking different values in Disentangle repo
+# TODO: check if any bool logic can be removed
+class VaeDatasetConfig(BaseModel):
```

Comment:
But at the same time, if it makes more sense to keep all the parameters together to initialise the main class, then that's also fine.

Reply:
Yes, you are totally right. I decided to put everything in the same place for now so it's easier to identify which parameters are used, since there are a lot of them. But in the future, it is definitely a good idea to leave only the user-customizable ones. 

About the data split type - I think there should be separate dataset classes for training and validation, and their instantiation should be a part of the training loop logic. But for now, it's left as it is not to mess anything up.

Reply:
Makes sense! I guess the (far future 😅) eventual aim is to move towards unified datasets across careamics anyway (although the careamics datasets need some work as well).